test/blackbox/common/BlackboxTestsDiscovery.cpp

```diff
@@ -542,6 +542,73 @@ TEST(Discovery, EndpointRediscoveryWithTransientLocalData)
     std::this_thread::sleep_for(std::chrono::seconds(1));
 }
 
+// Regression for bug #12127
+TEST(Discovery, EndpointRediscoveryWithTransientLocalData_2)
+{
+    // Both writer and reader will be RELIABLE, TRANSIENT_LOCAL, KEEP_LAST 10
+    // Partcipant lease duration will be small on both
+    // Writer will send 2 samples, but the reader will not take them
```

Comment:
This test is passing if cherry-picked into master so it is not properly testing the regression (I have run the `BlackboxTests_FastRTPS.Discovery.EndpointRediscoveryWithTransientLocalData_2` using the FastRTPS API)